src/functions.rs

```diff
@@ -141,6 +141,15 @@ fn resolve_event_ptrs<En: ClNullEventPtr, Ewl: ClWaitListPtr>(wait_list: Option<
     (wait_list_len, wait_list_ptr, new_event_ptr)
 }
 
+/// Converts a slice of Mem to a vec of cl_mem
+fn resolve_mem_ptrs(buffers: &[Mem]) -> Vec<cl_mem>{
```

Comment:
The idiomatic way to convert a slice to a Vec is is simply to use [`::to_owned`](https://doc.rust-lang.org/std/borrow/trait.ToOwned.html#tymethod.to_owned).

This function is also unnecessary, There is no need to convert a slice of `Mem`s into a Vec, OpenCL just wants a pointer to the start of the slice which we can get just using [`::as_ptr`](https://doc.rust-lang.org/std/primitive.slice.html#method.as_ptr) (noted below).